### ISSUE TITLE
Difference between training_outputs  with the input of training_epoch_end

### ISSUE THREAD

**Issue by lihui-w** [Lightning-AI/lightning#12616]
## 🐛 Bug

list[the output of the train_step] is different to the input of the training_epoch_end

This is my code of `training step`
```python
def training_step(self, batch, batch_idx):
		cfg = self.cfg
		x = batch[cfg.keys[0]]
		y = batch[cfg.keys[1]]
		y_hat = self.net(x)
                loss = self.loss_func(y_hat,y)
                print(loss)  #1
		return {'loss': loss}
   ```
   
   ```python
   def training_epoch_end(self, outputs):
		print(outputs)  #2
   ```
When I use GPU training the model, I have found that the answer printed is very different, and the later is nearly half of the former.

cc @justusschock @awaelchli @akihironitta @rohitgr7 @carmocca @borda @ananthsub @ninginthecloud @jjenniferdai

**Comment by rohitgr7** [comment]
are you seeing this with DP strategy?

**Comment by lihui-w** [comment]
yes, this influenced? with one GPU and two GPU, the problem is same.

**Comment by lihui-w** [comment]
Just now, i have tried DDPstragy with `trainer(gpus=[0])`. In the case, the loss is still half.

**Comment by rohitgr7** [comment]
can you share a minimal example showing this issue?
you can use https://colab.research.google.com/github/PyTorchLightning/pytorch-lightning/blob/master/pl_examples/bug_report/bug_report_model.ipynb for reference.

**Comment by lihui-w** [comment]
Thank you, I have found the reason while it's half, 
because I have using `trainer(accumulate_grad_batches=2,)`, it cause the outputs half of the outputs.

**Comment by rohitgr7** [comment]
`accumulate_grad_batches`shoulddon't have any effect on this. There must be something else.

**Comment by carmocca** [comment]
It's likely caused by this https://github.com/PyTorchLightning/pytorch-lightning/blob/184518c2fab188a9679a5b9d73ba95e3a8097280/pytorch_lightning/loops/optimization/optimizer_loop.py#L89 where normalize is `accumulate_grad_batches` https://github.com/PyTorchLightning/pytorch-lightning/blob/184518c2fab188a9679a5b9d73ba95e3a8097280/pytorch_lightning/loops/optimization/optimizer_loop.py#L436-L438

**Comment by rohitgr7** [comment]
oh okay... I thought the total length of outputs is half with accumulation.
